### PR TITLE
[Hardware][Ascend] Add optional unified SP attention executor

### PR DESCRIPTION
**Pull request by lynxxxi**
## Purpose

MindIE-SD provides a single-call `usp_attention` API that can own Ulysses communication, K/V gathering, and local FlashAttention execution. vLLM-Omni currently performs sequence-parallel communication before entering the NPU FlashAttention implementation, so the integration must run before those native collectives.

This PR adds a small, opt-in, platform-owned unified sequence-parallel (USP) executor:

- adds only the technical `DiffusionParallelConfig.enable_usp` switch (no new CLI or tuning parameters);
- keeps Ulysses/Ring topology in the existing `ulysses_degree` and `ring_degree` fields;
- keeps the public attention backend as `FLASH_ATTN`;
- exposes a generic platform hook from the portable attention layer;
- keeps the MindIE-SD import, ABI mapping, and structured errors inside the Ascend platform implementation;
- calls `mindiesd.layers.usp.usp_attention(q, k, v, ...)` once before native SP communication;
- passes Omni's Ulysses process group as `ulysses_group` and its **ring process group** as `kv_gather_group`;
- leaves chunking, communication dtype, overlap, and FA dispatch at platform defaults;
- falls back to the unchanged native Ulysses/Ring path when the optional API is unavailable or raises `USPError`.

The initial eligibility boundary is intentionally conservative: Ascend NPU, strict non-causal dense `FLASH_ATTN`, default softmax scale, standard BSND/SP dimensions, and no joint/masked/paged/UAA-specific semantics.

Companion RFC: #7229.

## Test Plan

- Unit contract tests cover platform selection, group mapping (including `kv_gather_group == ring_group`), delegation before native collectives, inactive-SP behavior, unsupported-semantics fallback, and exception handling.
- Passed locally:
  - Ruff check and format on all changed Python files.
  - `python -m compileall` on changed production modules.
  - pre-commit debug/end-of-file/line-ending/whitespace/typos/test-mark/SPDX/forbidden-import/torch.cuda hooks.
  - isolated Ascend USP contract smoke with a fake MindIE module.
  - `git diff --check`.
- Environment gaps:
  - Full pytest collection requires PyTorch and the complete Linux vLLM runtime; this Windows Python 3.14 environment has no PyTorch.
  - Ascend multi-rank numerical and performance validation requires an NPU environment with the matching MindIE-SD build.
  - mypy reports four pre-existing errors in `platforms/interface.py` and `attention/layer.py`; the new NPU USP module and tests introduce no additional mypy errors.

**vLLM Version:** `d4004455d2`

**vLLM-Omni Commit:** base `30d6a0b4e`, current PR head `3c8d9f81b`

## Test Result

Static checks and the isolated ABI contract pass. This PR remains a draft pending RFC feedback and Ascend multi-rank validation; it makes no measured performance claim yet.

### CODE REVIEW

File: vllm_omni/diffusion/attention/mindiesd_usp.py [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+
+class SequenceParallelGroups(Protocol):
+    """Process groups owned by vLLM-Omni's SP coordinator."""
+
+    ulysses_group: object
+    ring_group: object
+
+
+@dataclass(frozen=True, slots=True)
+class MindIESDUSPOptions:
```
Comment (hsliuustc0106) [inline vllm_omni/diffusion/attention/mindiesd_usp.py]: MindIESD is not a technical name for SP which is not allowed to add in this way

Comment (lynxxxi) [reply vllm_omni/diffusion/attention/mindiesd_usp.py]: ok

Comment (lynxxxi) [reply vllm_omni/diffusion/attention/mindiesd_usp.py]: Addressed in 3c8d9f81b. I removed MindIE-SD from the generic SP abstraction: the public config is now the technical `enable_usp` switch, `Attention` calls a platform-owned `build_diffusion_usp_executor`, and the provider-specific ABI/import handling lives under `platforms/npu/usp.py` as `AscendUSPExecutor`. Existing `ulysses_degree`/`ring_degree` remain the topology terms, with `kv_gather_group` mapped to `ring_group`.
